cohorts/load.py

```diff
@@ -397,14 +412,16 @@ def is_lambda(func):
             for key, value in on.iteritems():
                 col, df = apply_func(on=value, col=key, df=df)
                 cols.append(col)
-            return (cols, df)
         if type(on) == list:
             cols = []
             for i, elem in enumerate(on):
                 col = elem.__name__ if not is_lambda(elem) else "column_%d" % i
                 col, df = apply_func(on=elem, col=col, df=df)
                 cols.append(col)
-            return (cols, df)
+
+        if (rename_cols):
+            df = df.rename(columns=_strip_column_names(df.columns, keep_paren_contents=keep_paren_contents))
```

Comment:
Sorry for the confusing nature of this code! I think the `DataFrame` itself might get column-renamed while the `cols` variable won't, which could lead to issues.

I think `cols = _strip_column_names(cols, keep_paren_contents=keep_paren_contents)` would address that?